pkg/grpc/server_test.go

```diff
@@ -105,7 +115,17 @@ func TestClose(t *testing.T) {
 	t.Run("test close with api logging disabled", func(t *testing.T) {
 		port, err := freeport.GetFreePort()
 		require.NoError(t, err)
-		serverConfig := NewServerConfig("test", "127.0.0.1", port, []string{"127.0.0.1"}, "test", "test", 4, "", 4, false)
+		serverConfig := ServerConfig{
+			AppID:              "test",
+			HostAddress:        "127.0.0.1",
+			Port:               port,
+			APIListenAddresses: []string{"127.0.0.1"},
+			NameSpace:          "test",
+			TrustDomain:        "test",
+			MaxRequestBodySize: 4,
+			ReadBufferSize:     4,
+			EnableAPILogging:   false,
+		}
 		a := &api{}
 		server := NewAPIServer(a, serverConfig, config.TracingSpec{}, config.MetricSpec{}, config.APISpec{}, nil)
```

Comment:
Out of curiosity, why not include the tracing, metrics and api specs along with the proxy, in the `ServerConfig` struct?

Reply:
I have ported this 1:1 so this is what it was doing before. Can change it however if needed.